graphql/src/graphql_schema.ml

```diff
@@ -1089,28 +1161,97 @@ end
     | `Resolve_error of string
     | `Validation_error of string
     | `Mutations_not_configured
-    | `Subscriptions_not_implemented
+    | `Subscriptions_not_configured
     | `No_operation_found
     | `Operation_name_required
     | `Operation_not_found
   ]
 
-  let execute_operation : 'ctx schema -> 'ctx execution_context -> fragment_map -> variable_map -> Graphql_parser.operation -> (Yojson.Basic.json * string list, [> execute_error]) result Io.t =
+  type execute_operation =
+    [ `Response of Yojson.Basic.json
+    | `Stream of (Yojson.Basic.json, Yojson.Basic.json) result Io_stream.t]
+
+  let data_to_json = function
+    | data, [] -> `Assoc ["data", data]
+    | data, errors ->
+        let errors = List.map error_to_json errors in
+        `Assoc [
+          "data", data;
+          "errors", `List errors]
+
+  let error_response err =
+    `Assoc [
+      "errors", `List [
+        error_to_json err
+      ]
+    ]
+
+  let to_response = function
+    | Ok response as res -> res
+    | Error `No_operation_found ->
+        Error (error_response "No operation found")
+    | Error `Operation_not_found ->
+        Error (error_response "Operation not found")
+    | Error `Operation_name_required ->
+        Error (error_response "Operation name required")
+    | Error `Subscriptions_not_configured ->
+        Error (error_response "Subscriptions not configured")
+    | Error `Mutations_not_configured ->
+        Error (error_response "Mutations not configured")
+    | Error `Validation_error err ->
+        Error (error_response err)
+    | Error (`Argument_error err)
+    | Error (`Resolve_error err) ->
+        let `Assoc errors = error_response err in
+        Error (`Assoc (("data", `Null)::errors))
+
+  let subscribe : type ctx src. ctx execution_context -> src -> (ctx, src) subscription_field -> Graphql_parser.field -> ((Yojson.Basic.json, Yojson.Basic.json) result Io_stream.t, [`Argument_error of string | `Resolve_error of string]) result Io.t
+  =
+    fun ctx src (SubscriptionField subs_field) field ->
+      let open Io.Infix in
+      let subscriber = subs_field.subscribe ctx.ctx src in
+      begin match Arg.eval_arglist ctx.variables subs_field.args field.arguments subscriber with
+      | Ok source_stream -> Io.ok (
+          Io_stream.map_s (fun value ->
+              present ctx value field subs_field.typ
+              |> Io.Result.map ~f:(fun (data, errors) ->
+                (data_to_json (`Assoc [field.name, data], errors)))
```

Comment:
Should this be `alias_or_name field`?

Reply:
Definitely. When I read [this](http://facebook.github.io/graphql/June2018/#sec-Source-Stream) section of the spec, I interpreted `Note: This value is unaffected if an alias is used.` as not returning an alias, but that's not what they mean. That section is only about validation and finding the subscription field.